### PR TITLE
Feature: Added "Size on disk" to property to the Properties window

### LINKED ISSUE #3930: Feature: Add size on disk info to the properties dialog for better cloud management
**Is your feature request related to a problem? Please describe.**
I have always used the files information on windows files explorer to check size on disk, so that I can manage my files either to delete or free up space (keep in OneDrive), and it would be nice to see the same feature on this project.

**Describe the solution you'd like**
Enable to compute and view size on disk, rather than just size when viewing files properties.

**Describe alternatives you've considered**
not sure

**Additional context**
This is an example from windows files explorer:
image

and this is from current files app:
image

(note that this is the same file)

Thank you!

### PR DESCRIPTION
**Pull request by manfromarce**
This PR adds Size on disk to the Properties window like Windows Explorer.

**Resolved / Related Issues**
   Should close #3930.

**Validation**
How did you test these changes?
- [x] Did you build the app and test your changes?
- [ ] Did you check for accessibility? You can use Accessibility Insights for this.
- [ ] Did you remove any strings from the en-us resource file?
   - [ ] Did you search the solution to see if the string is still being used? 
- [ ] Did you implement any design changes to an existing feature?
   - [ ] Was this change approved?
- [x] Are there any other steps that were used to validate these changes? 
I just checked Properties for different types of items (file, directory, link, recycle bin item, zipped item)

**Screenshots (optional)**
Immagine 2023-06-28 152944

### CODE REVIEW

**Comment by yaira2** [comment]
> Are there any other steps that were used to validate these changes

Thank you for the PR. Can you fill this out this section? It helps accelerate the testing and review process.

File: src/Files.App/Data/Models/SelectedItemsPropertiesViewModel.cs [hunk 1]
```diff
@@ -165,6 +172,13 @@ public bool ItemSizeVisibility
 			set => SetProperty(ref itemSizeVisibility, value);
 		}
 
+		private bool itemSizeOnDiskVisibility = false;
+		public bool ItemSizeOnDiskVisibility
```
Comment (yaira2) [inline src/Files.App/Data/Models/SelectedItemsPropertiesViewModel.cs:176]: I think we can avoid this by creating a converter that's false when the string is empty.

Comment (manfromarce) [reply src/Files.App/Data/Models/SelectedItemsPropertiesViewModel.cs:176]: Ok, I will change this

File: src/Files.App/ViewModels/Properties/Items/CombinedProperties.cs [hunk 2]
```diff
@@ -65,15 +65,20 @@ public override async Task GetSpecialProperties()
...
+			ViewModel.ItemSizeOnDiskVisibility = true;
 
 			ViewModel.FilesCount += List.Where(x => x.PrimaryItemAttribute == StorageItemTypes.File || x.IsArchive).ToList().Count;
 			ViewModel.FoldersCount += List.Where(x => x.PrimaryItemAttribute == StorageItemTypes.Folder && !x.IsArchive).ToList().Count;
 
 			long totalSize = 0;
 			long filesSize = List.Where(x => x.PrimaryItemAttribute == StorageItemTypes.File).Sum(x => x.FileSizeBytes);
 			long foldersSize = 0;
+			long totalSizeOnDisk = 0;
+			long filesSizeOnDisk = List.Where(x => x.PrimaryItemAttribute == StorageItemTypes.File).Sum(x => NativeFileOperationsHelper.GetFileSizeOnDisk(x.ItemPath) ?? 0);
```
Comment (QuaintMako) [inline src/Files.App/ViewModels/Properties/Items/CombinedProperties.cs:77]: Formatting it by going to the next line between the linq methods would help the readability.

**Review by QuaintMako — COMMENTED** [review]
Code looks pretty clean and like it does its job. I cannot test it for now though.

**Comment by QuaintMako** [comment]
@yaira2 Should this be add to the Preview Pane as well?

File: src/Files.App/Views/Properties/GeneralPage.xaml
Comment (0x5bfa) [inline src/Files.App/Views/Properties/GeneralPage.xaml:1]: If you're not using XAML Styler, please use it to follow our code guidelines.

- VS2017/VS2019, download from [XAMLStyler](https://marketplace.visualstudio.com/items?itemName=TeamXavalon.XAMLStyler).
- VS2022, download from [XAMLStyler2022](https://marketplace.visualstudio.com/items?itemName=TeamXavalon.XAMLStyler2022).

Comment (manfromarce) [reply src/Files.App/Views/Properties/GeneralPage.xaml:1]: I've installed it but I don't know how to apply the style

**Comment by yaira2** [comment]
> Should this be add to the Preview Pane as well?

Not right now

File: src/Files.App/Views/Properties/GeneralPage.xaml
Comment (yaira2) [reply src/Files.App/Views/Properties/GeneralPage.xaml:1]: @manfromarce it should be applied to whichever xaml files were modified.

File: src/Files.App/Data/Models/SelectedItemsPropertiesViewModel.cs [hunk 3]
```diff
@@ -193,6 +207,13 @@ public bool ItemSizeProgressVisibility
 			set => SetProperty(ref itemSizeProgressVisibility, value);
 		}
 
+		private bool itemSizeOnDiskProgressVisibility = false;
+		public bool ItemSizeOnDiskProgressVisibility
```
Comment (yaira2) [inline src/Files.App/Data/Models/SelectedItemsPropertiesViewModel.cs:211]: Is this still used?

**Review by QuaintMako — APPROVED** [review]
LGTM and works as expected.

Some conflicts on the merge will need to be fixed though.

**Review by 0x5bfa — APPROVED** [review]
lgtm

**Comment by yaira2** [comment]
@manfromarce can you resolve the merge conflict?

**Comment by manfromarce** [comment]
> @manfromarce can you resolve the merge conflict?

It should be able to merge now

**Pull request merged**